cfme/cloud/keypairs.py

```diff
@@ -128,8 +128,12 @@ class KeyPair(BaseEntity, WidgetasticTaggable):
 
     def delete(self, cancel=False, wait=False):
         view = navigate_to(self, 'Details')
-        view.toolbar.configuration.item_select('Remove this Key Pair',
-                                               handle_alert=(not cancel))
+        if self.appliance.version < "5.8":
+            view.toolbar.configuration.item_select('Remove this Key Pair',
+                                                   handle_alert=(not cancel))
+        else:
+            view.toolbar.configuration.item_select('Remove this Key Pair from Inventory',
+                                                   handle_alert=(not cancel))
```

Comment:
**OPTIONAL**: You can get rid of the parentheses around `not cancel`

Reply:
Done.